openforcefield/utils/toolkits.py

```diff
@@ -1569,14 +1572,16 @@ def from_file(self,
                 mols.append(mol)
 
         elif (file_format == 'PDB'):
-            raise Exception(
-                "RDKit can not safely read PDBs on their own. Information about bond order and aromaticity "
-                "is likely to be lost.")
+            # raise Exception(
+            #     "RDKit can not safely read PDBs on their own. Information about bond order and aromaticity "
+            #     "is likely to be lost.")
             # TODO: See if we can implement PDB+mol/smi combinations to get complete bond information.
+            #  testing to see if we can make a molecule from smiles and then use the PDB conformer as the geometry
+            #  and just reorder the molecule
             # https://github.com/openforcefield/openforcefield/issues/121
-            # rdmol = Chem.MolFromPDBFile(file_path, removeHs=False)
-            # mol = Molecule.from_rdkit(rdmol)
-            # mols.append(mol)
+            rdmol = Chem.MolFromPDBFile(file_path, removeHs=False)
+            mol = Molecule.from_rdkit(rdmol)
+            mols.append(mol)
```

Comment:
Let's make sure to roll this change back -- Users should run a new function`molecule.from_pdb_and_smiles` (which requires a SMILES) to load from PDB. Inside that function, it could run code similar to here [1], specifically using `AssignBondOrdersFromTemplate`. Once the RDMol is prepared using that, we could just run `Molecule.from_rdkit` to make the OFFMol.

As you said, basically the only thing the PDB is used for in this process will be to get a conformer of the molecule (but guaranteeing the correct indexing for that geometry)

[1] https://practicalcheminformatics.blogspot.com/2018/09/assigning-bond-orders-to-pdb-ligands.html

Reply:
I am a little worried that this function might not transfer the stereochemistry which is assigned when making the molecule from SMILES. It might be better to approach it the other way around make a molecule from SMILES and reindex it to match the PDB order then just transfer the conformation?

Reply:
Ahhh, so RDKit's `AssignBondOrdersFromTemplate` doesn't pull in stereochemistry. Darn.

It looks like there's a PR to add stereo to that function, but it's been parked for 3 years. https://github.com/rdkit/rdkit/pull/886

Ok, so thinking of alternate solutions...

### Option 1
Most directly, we could just use `AssignBondOrdersFromTemplate`, and then run `Chem.AssignStereoFrom3D` on the resulting rdmol, similar to [1]. In this case, I think we'd ignore stereo from the SMILES and just trust whatever's in the geometry. It's... not perfect, but it'd work.

### Option 2 
> reindex it to match the PDB order then just transfer the conformation?

I see, but this would require an initial source of connectivity, no? Or we could do like using `simtk.openmm.app.PDBFile` to load the PDB, then convert it to a networkX graph, and then do the matching from there. That could be nice, since then we'd guarantee the user the stereochemistry from the SMILES.

I'll let you pick which one. 2) seems a bit more difficult, but marginally better in the long run.

[1] https://github.com/openforcefield/openforcefield/blob/master/openforcefield/utils/toolkits.py#L1551-L1553

Reply:
So long as the PDB has the connect terms at the bottom I can make a molecule from this and turn it into a networkx graph then pull in the conformation from the molecule and the mapping by doing an isomorphic check see here https://github.com/openforcefield/openforcefield/pull/472#discussion_r365340467